OpenRA.Mods.Common/Widgets/ProductionPaletteWidget.cs

```diff
@@ -364,44 +366,64 @@ bool HandleLeftClick(ProductionItem item, ProductionIcon icon, int handleCount,
 
 		bool HandleRightClick(ProductionItem item, ProductionIcon icon, int handleCount)
 		{
-			if (item == null)
+			if (item == null && CurrentQueue is not BulkProductionQueue)
 				return false;
 
-			Game.Sound.PlayNotification(World.Map.Rules, World.LocalPlayer, "Sounds", ClickSound, null);
-
-			if (CurrentQueue.Info.DisallowPaused || item.Paused || item.Done || item.TotalCost == item.RemainingCost || !item.Started)
+			if (item != null)
 			{
-				// Instantly cancel items that haven't started, have finished, or if the queue doesn't support pausing
-				Game.Sound.PlayNotification(World.Map.Rules, World.LocalPlayer, "Speech", CurrentQueue.Info.CancelledAudio, World.LocalPlayer.Faction.InternalName);
-				TextNotificationsManager.AddTransientLine(World.LocalPlayer, CurrentQueue.Info.CancelledTextNotification);
+				Game.Sound.PlayNotification(World.Map.Rules, World.LocalPlayer, "Sounds", ClickSound, null);
 
-				World.IssueOrder(Order.CancelProduction(CurrentQueue.Actor, icon.Name, handleCount));
+				if (CurrentQueue.Info.DisallowPaused || item.Paused || item.Done || item.TotalCost == item.RemainingCost || !item.Started)
+				{
+					// Instantly cancel items that haven't started, have finished, or if the queue doesn't support pausing
+					Game.Sound.PlayNotification(World.Map.Rules, World.LocalPlayer, "Speech", CurrentQueue.Info.CancelledAudio, World.LocalPlayer.Faction.InternalName);
+					TextNotificationsManager.AddTransientLine(World.LocalPlayer, CurrentQueue.Info.CancelledTextNotification);
+
+					World.IssueOrder(Order.CancelProduction(CurrentQueue.Actor, icon.Name, handleCount));
+				}
+				else
+				{
+					// Pause an existing item
+					Game.Sound.PlayNotification(World.Map.Rules, World.LocalPlayer, "Speech", CurrentQueue.Info.OnHoldAudio, World.LocalPlayer.Faction.InternalName);
+					TextNotificationsManager.AddTransientLine(World.LocalPlayer, CurrentQueue.Info.OnHoldTextNotification);
+
+					World.IssueOrder(Order.PauseProduction(CurrentQueue.Actor, icon.Name, true));
+				}
+
+				return true;
 			}
-			else
-			{
-				// Pause an existing item
-				Game.Sound.PlayNotification(World.Map.Rules, World.LocalPlayer, "Speech", CurrentQueue.Info.OnHoldAudio, World.LocalPlayer.Faction.InternalName);
-				TextNotificationsManager.AddTransientLine(World.LocalPlayer, CurrentQueue.Info.OnHoldTextNotification);
 
-				World.IssueOrder(Order.PauseProduction(CurrentQueue.Actor, icon.Name, true));
+			if (CurrentQueue is BulkProductionQueue bulkProductionQueue && !bulkProductionQueue.HasDeliveryStarted())
```

Comment:
Can we move this up to the start of the method? Then we can keep the
```
if (item == null)
	return false;
```

and the rest of the method unchanged.